test/AndcultureCode.CSharp.Sitefinity.Core.Tests/Unit/Extensions/DynamicContentExtensionsTests.cs

```diff
@@ -91,5 +99,48 @@ public void MapTo_Given_Properties_Dont_Match_Returns_Class_With_Property_Values
             mappedContent.TestPropertyWithDifferentName.ShouldBe(default(long));
             mappedContent.TestDateTimeOffsetProperty.ShouldBe(default(DateTimeOffset?));
         }
+
+        [Fact]
+        public void MapTo_Given_Enum_Type_With_Number_Value_Properties_Match_Returns_Class_With_Property_Enum_Value_Set()
+        {
+            // Arrange
+            var content = new TestDynamicClass();
+
+            content.TestEnumProperty = TestEnum.TestEnumWithNumber.GetHashCode().ToString();
+
+            // Act
+            var mappedContent = content.MapTo<TestValidClass>();
+
+            // Assert
+            mappedContent.TestEnumProperty.ShouldBe(TestEnum.TestEnumWithNumber);
+        }
+
+        [Fact]
+        public void MapTo_Given_Enum_Type_Without_Number_Value_Properties_Match_Returns_Class_With_Property_Enum_Value_Set()
+        {
+            // Arrange
+            var content = new TestDynamicClass();
+
+            content.TestEnumProperty = TestEnum.TestEnumWithoutNumber.ToString();
+
+            // Act
+            var mappedContent = content.MapTo<TestValidClass>();
+
+            // Assert
+            mappedContent.TestEnumProperty.ShouldBe(TestEnum.TestEnumWithoutNumber);
+        }
+
+        [Fact]
+        public void MapTo_Given_Enum_Type_With_Null_Property_Value_Does_Not_Throw_Exception()
+        {
+            // Arrange
+            var content = new TestDynamicClass();
+
+            content.TestEnumProperty = null;
+
+            // Act & Assert
+            var mappedContent = content.MapTo<TestValidClass>(); 
```

Comment:
Should there be an assertion here? I guess it would just fail if there _is_ an exception

Reply:
Correct.